Gh 2458 Update operation README with new sections (#2459)
gchq/Gaffer: core/operation/README.md

Operation implementations should override the close method and ensure all
closeable fields are closed.

The core Gaffer operations need to be registered in the `Store` class,
with their respective handlers. The `StoreTest` unit test also needs similar information.

> Operation implementations should override the close method and ensure all

Reviewer: stuplicate line

}
```

#### Operation Scores

For use with a `ScoreOperationChain`, some `Operation`s may require a custom
way of calculating an associated score, therefore an implementation of
the `ScoreResolver` interface may be required. There is a `DefaultScoreResolver`
to which the custom implementation should delegate, in a manner specific to the
new Operation. For more info, see [ScoreOperationChain](https://gchq.github.io/gaffer-doc/components/core/store.html#scoreoperationchain) and [ScoreOperationChainExample](https://gchq.github.io/gaffer-doc/getting-started/operations/scoreoperationchain.html).

#### Documentation

Class-level Javadoc should be provided for each `Operation`, giving a description of the functionality,
and any configuration information that may not immediately be obvious. Member-level Javadoc

> is not strictly necessary, but welcome if it would be beneficial 

Reviewer: ```suggestion
is not strictly necessary, but good practice for explanations/clarifications for complex methods.

```

> (e.g. explanations/clarifications for complex methods).

Reviewer: ```suggestion
```

> To assist users of the new Operation, it would be beneficial to provide documentation,

Reviewer: ```suggestion
To assist users of the new Operation, it is best practice to provide documentation,

```
